ppl/src/test/java/org/opensearch/sql/ppl/calcite/CalcitePPLRexTest.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.ppl.calcite;
+
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.test.CalciteAssert;
+import org.junit.Test;
+
+public class CalcitePPLRexTest extends CalcitePPLAbstractTest {
+  public CalcitePPLRexTest() {
+    super(CalciteAssert.SchemaSpec.SCOTT_WITH_TEMPORAL);
+  }
+
+  @Test
+  public void testRexBasicFieldExtraction() {
+    String ppl = "source=EMP | rex field=ENAME '(?<first>[A-Z]).*' | fields ENAME, first";
+    RelNode root = getRelNode(ppl);
+    String expectedLogical =
+        "LogicalProject(ENAME=[$1], first=[REX_EXTRACT($1, '(?<first>[A-Z]).*', 1)])\n"
+            + "  LogicalFilter(condition=[REGEXP_CONTAINS($1, '([A-Z]).*')])\n"
```

Comment:
why include LogicalFilter?

Reply:
fixed, this was coming from stalled test before we remove the filter operator. good catch